hw/ip/otbn/dv/uvm/otbn_sim_cfg.hjson

```diff
@@ -53,20 +55,34 @@ name:
   uvm_test: otbn_base_test
   uvm_test_seq: otbn_base_vseq
 
-  // The directory where ELF binaries should be found.
-  //
-  // TODO: This needs deciding upon. Eventually, it might be nice to allow this
-  //       to be overridden on the command line ("I've got a bunch of ELF files
-  //       that stress my new feature; please run the test suite with them")
+  // A build mode that sets up the toolchain. Run meson via
+  // meson_init.sh to generate the .env file that describes our toolchain.
+  build_modes: [
+    {
+      name: otbn_build_env_mode
+      pre_build_cmds: ["cd {proj_root} && BUILD_ROOT={build_dir} ./meson_init.sh"]
+    }
+  ]
+
+  // The default place to find and build ELF binaries. If you have
+  // some pre-built binaries, you can run the simulation with them by using
+  // --run-opts to override the otbn_elf_dir plusarg in run_opts.
   otbn_elf_dir: "{run_dir}/build-out"
+
   run_opts: ["+otbn_elf_dir={otbn_elf_dir}"]
 
-  // OTBN run modes.
+  // A run mode that runs the random instruction generator and builds the
+  // resulting binaries in {otbn_elf_dir}. If you override the otbn_elf_dir
+  // plusarg with --run-opts, we'll still build the binaries (but will ignore
+  // them).
+  gen_binaries: "{proj_root}/hw/ip/otbn/dv/uvm/gen-binaries.py"
+  otbn_obj_dir: "{build_dir}/build-out"
   run_modes: [
     {
       name: build_otbn_rig_binaries_mode
-      pre_run_cmds: ["BUILD_ROOT={run_dir} cd {proj_root} && ./meson_init.sh",
-                     "{proj_root}/hw/ip/otbn/dv/uvm/gen-binaries.py --seed {seed} {otbn_elf_dir}"]
+      pre_run_cmds: [
+        "{gen_binaries} --seed {seed} --obj-dir {otbn_obj_dir} {otbn_elf_dir}"
+      ]
```

Comment:
Build modes can have runtime-y things such as `pre_run_cmds`. I dont think this run mode will be used without the `otbn_build_env_mode`. So you can perhaps move the `pre_run_cmds` directly into that build_mode.

Reply:
I understand. The problem is that we build the testbench once and then only some of the tests need to build binaries, so I think this needs to remain split.

Reply:
Okay. You still need to enable this mode within the test specifications, right?

Reply:
Yep. I'm imagining that we're probably going to change the `gen-binaries.py` call for different tests (e.g. change the number of binaries that are generated), so I don't think the repetition is going to be too much of an issue.